activation/nipost.go

```diff
@@ -251,11 +263,19 @@ func (nb *NIPostBuilder) BuildNIPost(
 		return nil, fmt.Errorf("submitting to poets: %w", err)
 	}
 
-	// Phase 1: query PoET services for proofs
+	// Stage 1: query PoET services for proofs
+	if err := nb.identityStates.Set(signer.NodeID(), types.IdentityStateWaitForPoetRoundEnd); err != nil {
+		nb.logger.Warn("failed to switch identity state",
+			zap.Stringer("smesherID", signer.NodeID()),
+			zap.Error(err),
+		)
+	}
+
 	poetProofRef, membership, err := nipost.PoetProofRef(nb.localDB, signer.NodeID())
 	if err != nil && !errors.Is(err, sql.ErrNotFound) {
 		nb.logger.Warn("cannot get poet proof ref", zap.Error(err))
```

Comment:
As above, are we supposed to continue here, given that the error may be pretty much anything, e.g. a database failure?

Reply:
Same as above, and it is node db failure, it is in-memory storage (map) and it exists only to return state via API. There is only one possible error - if we are trying to make not allowed state switch. That's why I think, the registration mechanism should work as before.